src/components/AsyncTypeahead/AsyncTypeahead.js

```diff
@@ -0,0 +1,305 @@
+    },
+  }))
+
+  let className = styles.Option
+
+  if (itemIndex === currentActive) {
+    className = `${className} ${styles.ActiveOption}`
+  }
+  // Note the if an item is both active (you've navigated to it)
+  // and also selected (it was the previous selection), we want
+  // both of those classes as that has it's own unique affordance.
+  // See the AsyncTypeahead.module.scss for the details :)
+  if (itemIndex === selectedOption) {
+    className = `${className} ${styles.SelectedOption}`
+  }
+  return (
+    <li ref={innerRef}>
+      <button
+        className={className}
+        onClick={() => {
+          setSelectedOptionDelegate(itemIndex)
+          onChange(item)
+        }}
+      >
+        {item[dataKey]}
+      </button>
+    </li>
+  )
+})
+Item.displayName = 'Item'
+Item.propTypes = {
+  currentActive: PropTypes.number,
+  itemIndex: PropTypes.number,
+  item: PropTypes.object,
+  dataKey: PropTypes.string,
+  selectedOption: PropTypes.number,
+  setSelectedOptionDelegate: PropTypes.func,
+  onChange: PropTypes.func,
+}
+
+/**
+ * AsyncTypeahead is a component that allows you to make asynchronous API
+ * fetches, and then use the results (entities) to show suggested results
+ * as dropdown options. It debounces the captured input via the
+ * `useFetchEntities` hook which takes in your dependency injected `fetchCallback`
+ * allowing you to fetch for any arbitrary entity (be it Posts, Users, or whatever).
+ * In order to be flexible in this way, it also takes in the `dataKey` to parse the
+ * API data with e.g. if you a list of items and .name has the token you wish
+ * to display in the dropdown results, you'd pass in `dataKey="name"`.
+ *
+ * @public
+ *
+ * @return {JSX.Element}
+ */
+export const AsyncTypeahead = ({
+  renderInput,
+  value,
+  dataKey,
+  onChange,
+  fetchCallback,
+  minChars = 1,
+  placeholder = 'Search...',
+}) => {
+  let optionsRefs = []
+  const [searchString, setSearchString] = useState('')
+  const [showOptions, setShowOptions] = useState(false)
+  const [activeOption, setActiveOption] = useState(0)
+  const [selectedOption, setSelectedOption] = useState(-1)
+
+  const { entities, loading } = useFetchEntities({
+    searchString,
+    fetchEntities: fetchCallback,
+    delay: 300,
+  })
+
+  /**
+   * Has user typed min chars to show dropdown options
+   */
+  const hasMinChars = (inputValue) => {
+    return inputValue.length >= minChars
+  }
+
+  /**
+   * Sets state so that we'll show the dropdown options
+   */
+  const setShow = (e) => {
+    const inputValue = e.target.value
+    setShowOptions(hasMinChars(inputValue))
+  }
+
+  /**
+   * Sets the selected options and also active option
+   */
+  const setSelectedOptionDelegate = (index) => {
+    setSelectedOption(index)
+    setActiveOption(index)
+  }
+
+  /**
+   * Determines if the user's input value is shorter then the
+   * currently selected value object.
+   */
+  const termIsLessThenValue = (inputValue) => {
```

Comment:
I'm having a bit of a hard time wrapping my head around this logic and how it's used to clear the search.

What does `term` and value refer to in this case?

Reply:
Maybe a bad name, I meant it as in "search term is less then current value". I suppose value is probably getting overloaded too much 🤔 my comment is actually telling what's going on. If I have time I'll contemplate some better variable names to make this less confusing

Reply:
hopefully fixed in latest...thanks for feedback!